### PR TITLE
Add withPreventDoubleClick HoC

### PR DESCRIPTION
**Pull request by rickycodes**
**Description**

This makes it so you can't double click in Settings (this component might be useful elsewhere?)

**Before:**
(you could double click on a setting and it'll push the same thing to the navigation stack twice)

https://user-images.githubusercontent.com/675259/104971619-4d916f80-59bd-11eb-8cbe-f97a0ab8989a.mp4

**After:**
(accidentally double clicking only pushes one thing to the stack)

https://user-images.githubusercontent.com/675259/104971726-9d703680-59bd-11eb-9cc3-48d939cc1bd0.mp4

**Checklist**

* [x] Tests are included if applicable

### CODE REVIEW

File: app/components/UI/Button/index.js [hunk 1]
```diff
@@ -33,15 +33,15 @@ Button.propTypes = {
 	 * it can be a text node, an image, or an icon
 	 * or an Array with a combination of them
 	 */
-	children: PropTypes.any,
+	children: PropTypes.any.isRequired,
 	/**
 	 * Styles to be applied to the Button
 	 */
-	style: ViewPropTypes.style,
+	style: ViewPropTypes.style.isRequired,
```
Comment (wachunei) [inline app/components/UI/Button/index.js:40]: I would not mark this as required

File: app/components/UI/Button/index.js [hunk 2]
```diff
@@ -33,15 +33,15 @@ Button.propTypes = {
...
 	/**
 	 * Function to be called on press
 	 */
-	onPress: PropTypes.func
+	onPress: PropTypes.func.isRequired
```
Comment (wachunei) [inline app/components/UI/Button/index.js:44]: I would not mark this as require either (default RN Button does not require it). An example: we want to display a button inside a box but make the whole box touchable, in that case the button would not require to repeat the onPress

File: app/components/UI/WithPreventDoubleClick/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			this.props.onPress && this.props.onPress();
+		};
+
+		onPress = debounce(this.debouncedOnPress, 300, { leading: true, trailing: false });
+
+		render() {
+			return <WrappedComponent {...this.props} onPress={this.onPress} />;
+		}
+	}
+	const propTypes = {
```
Comment (wachunei) [inline app/components/UI/WithPreventDoubleClick/index.js:23]: I'd move this to top level scope

**Review by wachunei — APPROVED** [review]
LGTM

**Comment by rickycodes** [comment]
closing in favour of: https://github.com/MetaMask/metamask-mobile/pull/2191

**Pull request closed without merging**